netrd/dynamics/lotka_vottera.py

```diff
@@ -0,0 +1,135 @@
+"""
+lotka_vottera.py
+----------------
+
+Implementation to simulate a Lotka-Vottera model on a network.
+
+author: Chia-Hung Yang
+Submitted as part of the 2019 NetSI Collabathon.
+"""
+
+from netrd.dynamics import BaseDynamics
+import numpy as np
+import networkx as nx
+from numpy.random import unifrom, normal
+from scipy.integrate import ode
+
+
+class LotkaVottera(BaseDynamics):
+    def simulate(self, G, L, init=None, gr=None, cap=None, inter=None,
+                 dt=1e-2, stochastic=True, pertb=None):
+        """Simulate time series on a network from the Lotka-Vottera model.
+
+        The Lotka-Vottera model was designed to describe dynamics of species
+        abundances in an ecosystem. Species $i$'s abundance change per time is
+        $\frac{d X_i}{d t} = r_i * X_i * \left(1 - \frac{X_i}{K_i} + \sum_{j \neq i} W_{ij} \frac{X_j}{K_i}\right)$
+        where $r_i$ and $K_i$ are the growth rate and the carrying capacity of
+        species $i$ respectively, and $W_{ij}$ are the relative interaction
+        strength of species $j$ on $i$.
+
+        Params
+        ------
+        G (nx.Graph): Underlying ground-truth network of simulated time series
+                      which has $N$ nodes.
+
+        L (int): Length of time series.
+
+        init (np.ndarray): Length-$N$ 1D array of nodes' initial condition. If
```

Comment:
We don't do a lot of sanity-checking in general; I'm not sure it's worth the trouble to guarantee that all these parameters are of the appropriate dimensions when passed.

Thoughts, @tlarock?

Reply:
If we do this here, we should do it everywhere. Since we may be doing similar things elsewhere without doing a check, let's leave it as is for now. Users should read the docstring and my guess is that more often than not they will not specify initial conditions.